### PR TITLE
Replace custom in-memory logic with in-memory sqlite connection

### PR DESCRIPTION
**Pull request by prha**
### Summary & Motivation
Simplify the codebase.

### How I Tested These Changes
BK

### CODE REVIEW

File: python_modules/dagster/dagster/_core/storage/sqlite.py [hunk 1]
```diff
@@ -13,5 +13,9 @@ def create_db_conn_string(base_dir, db_name):
     return "sqlite:///{}".format("/".join(path_components + [db_file]))
 
 
+def create_in_memory_conn_string(db_name):
+    return f"sqlite:///file:{db_name}?mode=memory&uri=true"
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/storage/sqlite.py:17]: why not just `:memory:` ? looking at docs guessing shared cache / foreign keys, might be worth leaving a comment here

File: python_modules/dagster/dagster/_core/storage/runs/in_memory.py [hunk 2]
```diff
@@ -1,410 +1,54 @@
...
-            return False
-
-        return True
-
-    return _filter
+    def __init__(self, preload=None):
+        self._conn = self._create_connection()
 
+        self.migrate()
+        self.optimize()
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/storage/runs/in_memory.py:24]: ? isnt this a fresh DB everytime?

File: python_modules/dagster/dagster/_core/storage/event_log/in_memory.py [hunk 3]
```diff
@@ -1,46 +1,57 @@
...
-        self._logs = defaultdict(list)
-        self._handlers = defaultdict(set)
         self._inst_data = inst_data
-        self._wiped_asset_keys = defaultdict(float)
+        self._conn = self._create_connection()
+        self._handlers = defaultdict(set)
+        self._storage_id = 0  # mirror the storage id, to mimic watching cursors
+
+        self.reindex_events()
+        self.reindex_assets()
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/storage/event_log/in_memory.py:31]: do we need this?

**Review by alangenfeld — APPROVED** [review]
finally got around to it 👏 - good shit

File: python_modules/dagster/dagster/_core/storage/runs/in_memory.py (hunk 2 above)
Comment (prha) [reply python_modules/dagster/dagster/_core/storage/runs/in_memory.py:24]: We have some unit tests in place that check to see if the migrations and some of the secondary index values are set

File: python_modules/dagster/dagster/_core/storage/event_log/in_memory.py (hunk 3 above)
Comment (prha) [reply python_modules/dagster/dagster/_core/storage/event_log/in_memory.py:31]: We have some checks to make sure that new DBs will have the secondary indexes marked correctly (for optimized queries)

**Pull request merged**